src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -1,4 +1,56 @@
 package com.kodilla.ecommercee.domain;
 
+import com.kodilla.ecommercee.exception.ProductNotFoundInCartException;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.GeneratedValue;
+import javax.persistence.ManyToMany;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.CascadeType;
+import javax.persistence.JoinTable;
+import javax.persistence.JoinColumn;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table(name="CARTS")
 public class Cart {
+    private Long id;
+    private List<Product> products = new ArrayList<>();
+
+    public Cart() {};
+
+    public Cart(Long id, List<Product> productList) {
```

Comment:
W encji user nie leci w konstruktorze lista... tutaj leci :) pytanie czy ta różnica jest uzasadniona? Jeśli robimy konstruktor bezparametrowy i konstruktor, który ustawia wszystkie parametry to raczej to powinno być.Tak musimy stworzyć entity przez konstruktor i listę zasilić osobno pod wywołaniem konstruktora.

Reply:
https://github.com/pplutap/JDP-1901-01/blob/577296c5b85bebb3a28861205062d28a64aba48f/src/main/java/com/kodilla/ecommercee/domain/User.java